services/QuillLMS/engines/evidence/lib/evidence/evidence/open_ai/completion.rb

```diff
@@ -17,12 +17,12 @@ class Completion < Evidence::ApplicationService
         davinci: 'text-davinci-002'
       }
       BLANK = ''
-      STOP_TOKENS = [". ", ", "]
+      STOP_TOKENS = [". ", "; ", "? ", "! "] # max of 4 stop tokens
       MAX_COUNT = 128 # API has an undocument max of 128 for 'n'
 
       attr_accessor :response, :prompt, :temperature, :count, :model_key, :options_hash
 
```

Comment:
This line and the additions to stop tokens are the only ones that have an effect on current production code.